hooks/shotgun_launch_app.py

```diff
@@ -0,0 +1,85 @@
+# Copyright (c) 2019 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Hook for launching the published file in an app. The path to the app
+in different platforms is defined in the configuration.
+
+This hook can be used in conjunction with the `shotgun_get_valid_published_file`
+hook in order to be sure a proper published file with a valid extension is provided
+to the app.
+
+"""
+
+import os
+import sys
+import sgtk
+from tank import TankError
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class LaunchApp(HookBaseClass):
+    """
+    A Hook to open a file in a predefined app.
+    """
+
+    def execute(self, published_file, **kwargs):
+        """
+        Try to launch the defined application for the given
+        published file's path.
+
+        :param published_file: the published file.
+
+        :raises: TankError, PublishPathNotDefinedError, PublishPathNotSupported
+        """
+        # Will raise an error if the path is not defined or cannot be resolved
+        # The app will take care of it.
+        publish_path = self.get_publish_path(published_file)
+        self.logger.debug("Launching app for file %s" % publish_path)
+        self._launch_app(publish_path)
+
+    def _launch_app(self, path):
+        """
+        Launches an app based on config settings.
+        We assume that the path to the file is just passed as a param to the app.
+        This seems to be standard for most apps.
+        """
+        # get the setting
+        system = sys.platform
+        try:
+            app_setting = {"linux2": "app_path_linux",
+                           "darwin": "app_path_mac",
+                           "win32": "app_path_windows"}[system]
+            app_path = self.parent.get_setting(app_setting)
+            if not app_path:
+                raise KeyError()
+        except KeyError:
```

Comment:
This seems unnecessary. Why not remove the try/except and just : 

```python
if not app_path:
    raise TankError("Cannot find app path for platform '%s'." % system)
```

Reply:
actually it can raise a KeyError above, but I'll fix that.